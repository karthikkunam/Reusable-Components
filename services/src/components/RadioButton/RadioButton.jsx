import React, { useState } from 'react';
import PropTypes from 'prop-types';

const RadioButton = props => {
  const [value, setValue] = useState(props.checked || props.defaultValue);
  const { radioOptions, onClick, className, checked } = props;

  const handleRadioClick = option => {
    if (option.disabled) {
      return;
    }
    onClick && onClick(option);
    setValue(option.value);
  };
  return (
    <fieldset className={`form-group radio mt-1 ${className}`}>
      {radioOptions &&
        radioOptions.length > 0 &&
        radioOptions.map((option, index) => {
          return (
            <label
              key={index}
              className={`radio-label ${option.disabled ? 'disabled' : ''}`}
              onClick={() => handleRadioClick(option)}>
              <span className="radio-text">{option.label}</span>
              <input type="radio" checked={option.value === value || checked ? 'checked' : ''} name="radio" />
              <span className="checkmark"></span>
            </label>
          );
        })}
    </fieldset>
  );
};
RadioButton.defaultProps = {
  className: '',
};

RadioButton.propTypes = {
  radioOptions: PropTypes.arrayOf({}).isRequired,
  defaultValue: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  checked: PropTypes.string,
};

export default RadioButton;
