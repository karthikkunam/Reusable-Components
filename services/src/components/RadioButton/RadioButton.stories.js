import React from 'react';
import RadioButton from '.';
import { action } from '@storybook/addon-actions';

export default {
  title: 'RadioButton',
  component: RadioButton,
};

const radioOptions = [
  {
    label: 'One',
    value: 1,
  },
  {
    label: 'Two',
    value: 2,
  },
  {
    label: 'Three',
    value: 3,
  },
];
const defaultRadioOptions = [
  {
    label: 'One',
    value: 1,
  },
  {
    label: 'Two',
    value: 2,
  },
  {
    label: 'Three',
    value: 3,
    disabled: true,
  },
];
const disabledRadio = [
  {
    label: 'One',
    value: 1,
    disabled: true,
  },
];

export const DefaultRadioButton = () => (
  <>
    <div> Radio Buttons</div>
    <RadioButton radioOptions={radioOptions} onClick={action('checked')} />
  </>
);
export const RadioButtonWithDefaultValue = () => (
  <>
    <div> Radio Buttons with default value</div>
    <RadioButton radioOptions={defaultRadioOptions} defaultValue={1} onClick={action('checked')} />
  </>
);
export const RadioButtonCheckedDisbaled = () => (
  <>
    <div> Radio Buttons (checked disabled)</div>
    <RadioButton checked={true} radioOptions={disabledRadio} />
  </>
);
export const RadioRightButtons = () => (
  <>
    <div> Radio Right Buttons</div>
    <RadioButton className="radio-right" radioOptions={radioOptions} onClick={action('checked')} />
  </>
);
