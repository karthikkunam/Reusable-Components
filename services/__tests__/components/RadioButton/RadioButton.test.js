import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from '../../../src/components/RadioButton';

describe('Radio Component', () => {
  it('renders correctly', () => {
    const props = {
      radioOptions: [{}],
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<RadioButton {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders handleRadioClick correctly', () => {
    const props = {
      radioOptions: [
        {
          label: 'One',
          value: 1,
        },
        {
          label: 'Two',
          value: 2,
        },
        {
          label: 'Three',
          value: 3,
          disabled: true,
        },
      ],
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<RadioButton {...props} />);
    component
      .find('label')
      .at(0)
      .simulate('click', { handleRadioClick: () => {}, setValue: () => {} });
    expect(component).toMatchSnapshot();
  });
  it('should return null when disabled is true onclick of handleRadioClick', () => {
    const props = {
      radioOptions: [
        {
          label: 'One',
          value: 1,
        },
        {
          label: 'Two',
          value: 2,
          disabled: true,
        },
      ],
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<RadioButton {...props} />);
    component
      .find('label')
      .at(1)
      .simulate('click', { option: { disabled: true } });
    expect(component).toMatchSnapshot();
  });
  it('renders Radio button with default value correctly', () => {
    const props = {
      radioOptions: [
        {
          label: 'One',
          value: 1,
        },
        {
          label: 'Two',
          value: 2,
        },
        {
          label: 'Three',
          value: 3,
          disabled: true,
        },
      ],
      defaultValue: 1,
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<RadioButton {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders Radio button with checked and disable', () => {
    const props = {
      radioOptions: [
        {
          label: 'One',
          value: 1,
        },
        {
          label: 'Two',
          value: 2,
        },
        {
          label: 'Three',
          value: 3,
          disabled: true,
          checked: true,
        },
      ],
      defaultValue: 1,
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<RadioButton {...props} />);
    expect(component).toMatchSnapshot();
  });
});
