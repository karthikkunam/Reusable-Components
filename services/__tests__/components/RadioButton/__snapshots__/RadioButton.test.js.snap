// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component renders Radio button with checked and disable 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      One
    </span>
    <input
      checked="checked"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label "
    key="1"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Two
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label disabled"
    key="2"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Three
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Radio Component renders Radio button with default value correctly 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      One
    </span>
    <input
      checked="checked"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label "
    key="1"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Two
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label disabled"
    key="2"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Three
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Radio Component renders correctly 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    />
    <input
      checked="checked"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Radio Component renders handleRadioClick correctly 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      One
    </span>
    <input
      checked="checked"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label "
    key="1"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Two
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label disabled"
    key="2"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Three
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Radio Component should return null when disabled is true onclick of handleRadioClick 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      One
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label disabled"
    key="1"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Two
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;
